### PR TITLE
[Merged by Bors] - feat: port Computability.TMToPartrec

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #3851

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3862
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3851~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Computability/TMToPartrec.lean [hunk 1]
```diff
@@ -1029,33 +1111,82 @@ def tr : Λ' → Stmt'
...
       goto fun s =>
         cond (natEnd s.iget) (Λ'.ret k) <| Λ'.clear natEnd main <| trNormal f (Cont'.fix f k)
-  | Λ'.ret cont'.halt => (load fun _ => none) <| halt
+  | Λ'.ret Cont'.halt => (load fun _ => none) <| halt
 #align turing.partrec_to_TM2.tr Turing.PartrecToTM2.tr
 
-/-- Translating a `cont` continuation to a `cont'` continuation simply entails dropping all the
-data. This data is instead encoded in `tr_cont_stack` in the configuration. -/
+/- Porting note: The equation lemma of `tr` simplifies to `match` structures. To prevent this,
+We replaces equation lemmas of `tr`. -/
```
Comment (jcommelin) [inline Mathlib/Computability/TMToPartrec.lean:1129]:
```suggestion
we replace equation lemmas of `tr`. -/
```

File: Mathlib/Computability/TMToPartrec.lean [hunk 2]
```diff
@@ -1114,134 +1243,137 @@ For example:
...
-@[simp]
 def K'.elim (a b c d : List Γ') : K' → List Γ'
   | K'.main => a
   | K'.rev => b
   | K'.aux => c
   | K'.stack => d
 #align turing.partrec_to_TM2.K'.elim Turing.PartrecToTM2.K'.elim
 
+/- Porting note: The equation lemma of `elim` simplifies to `match` structures. To prevent this,
+We replaces equation lemmas of `elim`. -/
```
Comment (jcommelin) [inline Mathlib/Computability/TMToPartrec.lean:1293]:
```suggestion
we replace equation lemmas of `elim`. -/
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**